cmd/kubeadm/app/util/kubeconfig/kubeconfig_test.go

```diff
@@ -345,3 +346,184 @@ func TestHasCredentials(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterFromKubeConfig(t *testing.T) {
+	tests := []struct {
+		name                string
+		config              *clientcmdapi.Config
+		expectedClusterName string
+		expectedCluster     *clientcmdapi.Cluster
+	}{
+		{
+			name: "cluster is empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "kubernetes",
+			},
+			expectedClusterName: "",
+			expectedCluster:     nil,
+		},
+		{
+			name: "cluster and currentContext are not empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"foo": {AuthInfo: "foo", Cluster: "foo"},
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expectedClusterName: "foo",
+			expectedCluster: &clientcmdapi.Cluster{
+				Server: "http://foo:8080",
+			},
+		},
+		{
+			name: "cluster is not empty and currentContext is not in contexts",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expectedClusterName: "",
+			expectedCluster:     nil,
+		},
+	}
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			got, got1 := GetClusterFromKubeConfig(rt.config)
```

Comment:
please also change to
```suggestion
			clusterName, cluster := GetClusterFromKubeConfig(rt.config)
```